### PR TITLE
ParaView: let paraview build its own verdict tpl

### LINKED ISSUE #30250: Installation issue: Praview needs Verdict
### Steps to reproduce the issue

```console
$ spack install paraview@master+qt+python3
```

Installing paraview crash during CMake because it does not find the Verdict lib.
Am I the only one to have this ?
To fix this, we could add ```'-DVTK_MODULE_USE_EXTERNAL_verdict=OFF'``` in the CMake args

@chuckatkins @danlipsa  @vicentebolea, do you agree ?

### Error message

```
1 error found in build log:
     225    Call Stack (most recent call first):
     226      VTK/CMake/vtkModule.cmake:5065 (vtk_module_find_package)
     227      VTK/CMake/vtkModule.cmake:4936 (vtk_module_third_party_external)
     228      VTK/ThirdParty/verdict/CMakeLists.txt:1 (vtk_module_third_party)
     229    
     230    
  >> 231    CMake Error at VTK/CMake/vtkModule.cmake:4471 (message):
     232      Could not find the Verdict external dependency.
     233    Call Stack (most recent call first):
     234      VTK/CMake/vtkModule.cmake:5065 (vtk_module_find_package)
     235      VTK/CMake/vtkModule.cmake:4936 (vtk_module_third_party_external)
     236      VTK/ThirdParty/verdict/CMakeLists.txt:1 (vtk_module_third_party)
     237    

See build log for details:
  /tmp/billae/spack-stage/spack-stage-paraview-master-ob4aut4wpm5cue2sl6ecxc5ubtphxztd/spack-build-out.txt
```

### Information on your system

* **Spack:** 0.18.0.dev0 (ac9bd3d72d7a4ed49e5bbdd3f07a4fdf91f8d0ce)
* **Python:** 3.6.9
* **Platform:** linux-ubuntu18.04-skylake
* **Concretizer:** clingo

### Additional information

_No response_

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

### PR DESCRIPTION
**Pull request by EthanS94**
@vicentebolea @danlipsa @chuckatkins I agree with @Billae proposed fix for their issue .

Fixes #30250

### CODE REVIEW

File: var/spack/repos/builtin/packages/paraview/package.py [hunk 1]
```diff
@@ -345,6 +345,9 @@ def nvariant_bool(feature):
             '-DBUILD_TESTING:BOOL=OFF',
             '-DOpenGL_GL_PREFERENCE:STRING=LEGACY']
 
+        if spec.satisfies('@master'):
```
Comment (danlipsa) [inline var/spack/repos/builtin/packages/paraview/package.py]: I would also include versions greater than 5.10.1

**Review by danlipsa — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/paraview/package.py (hunk 1 above)
Comment (EthanS94) [reply var/spack/repos/builtin/packages/paraview/package.py]: Can do. `@5.11:` is good?

**Comment by danlipsa** [comment]
+1

**Comment by Billae** [comment]
+1 =)

File: var/spack/repos/builtin/packages/paraview/package.py (hunk 1 above)
Comment (chuckatkins) [inline var/spack/repos/builtin/packages/paraview/package.py]: This should also apply to >= 5.11 so it dosn't need to be changed once the 5.11 release cycle starts.
```suggestion
        if '@5.11:,master' in spec:
```

Comment (chuckatkins) [reply var/spack/repos/builtin/packages/paraview/package.py]: That should be good.  i.e. `if '@5.11:,master' in spec:`

Comment (danlipsa) [reply var/spack/repos/builtin/packages/paraview/package.py]: @chuckatkins Why do we need to specify master? spack could do some git queries to see where master is located. That would be a nice improvement.

Comment (Billae) [reply var/spack/repos/builtin/packages/paraview/package.py]: I'm surprised too: I thought spack consider master > any release number

Comment (vicentebolea) [reply var/spack/repos/builtin/packages/paraview/package.py]: > I would also include versions greater than 5.10.1

I do not think that a patch release will change the VTK versions.

I found the commit introducing this at https://gitlab.kitware.com/vtk/vtk/-/commit/930d460e37c3993814b94298b956d2f2e0b47bc2

This commit was introduced to master here https://gitlab.kitware.com/paraview/paraview/-/merge_requests/5522, the master commit is https://gitlab.kitware.com/paraview/paraview/-/commit/0e42156a2b9b4d6066bb1510231073de723cf6b1.

I wonder if there is any way in Spack to say: if "after this particular commit"

if we do not have such tool perhaps we can consider enabling this for everything after 5.10.1 (inclusive, even 5.10.1 that does not need it) considering that this is harmless for 5.10.1.

Comment (danlipsa) [reply var/spack/repos/builtin/packages/paraview/package.py]: > I wonder if there is any way in Spack to say: if "after this particular commit"

I don't think there is a way to specify a SHA in spack. 

>  we can consider enabling this for everything after 5.10.1 (inclusive, even 5.10.1 that does not need it) considering that this is harmless for 5.10.1

That would be fine. The problem is that I think Chuck is right in saying that won't include master. I think it has to do with how versions are compared. Thinking about it, I think the condition "all versions >= certain version" should always include master.

Comment (danlipsa) [reply var/spack/repos/builtin/packages/paraview/package.py]: What about spec.satisfies('@5.10.2:')? That would include master according to https://spack.readthedocs.io/en/latest/packaging_guide.html#version-comparison

File: var/spack/repos/builtin/packages/paraview/package.py [hunk 2]
```diff
@@ -345,6 +345,9 @@ def nvariant_bool(feature):
             '-DBUILD_TESTING:BOOL=OFF',
             '-DOpenGL_GL_PREFERENCE:STRING=LEGACY']
 
+        if spec.satisfies('@5.11:'):
```
Comment (danlipsa) [inline var/spack/repos/builtin/packages/paraview/package.py:348]: Does this include master?

Comment (EthanS94) [reply var/spack/repos/builtin/packages/paraview/package.py:348]: @danlipsa It does. I am building `paraview@master` nightly with this patched in and it's getting it.

Comment (danlipsa) [reply var/spack/repos/builtin/packages/paraview/package.py:348]: Great. Then it is good to go for me. +1

**Comment by vicentebolea** [comment]
+1

File: var/spack/repos/builtin/packages/paraview/package.py (hunk 1 above)
Comment (vicentebolea) [reply var/spack/repos/builtin/packages/paraview/package.py]: @danlipsa 

> The most-recent development version of a package will usually be newer than any released numeric versions.

5.10.2 is not __released__ numeric version, I wonder if the choice of the word _release_ is a figure of speech, perhaps we could try to see this, your proposition would then be the most precise solution.

**Review by danlipsa — APPROVED** [review]

**Review by alalazo — APPROVED** [review]

File: var/spack/repos/builtin/packages/paraview/package.py (hunk 1 above)
Comment (danlipsa) [reply var/spack/repos/builtin/packages/paraview/package.py]: Yes, 5.11 works to do the same thing so there is no need for other versions. I was trying to come up with an earlier version than the current master but later than the released version.

**Pull request merged**